src/execution_plan/ops/op_aggregate.c

```diff
@@ -12,15 +12,151 @@
 #include "../../query_executor.h"
 #include "../../arithmetic/aggregate.h"
 
+static void _expandCollapsedNodes() {
+    AST *ast = AST_GetFromLTS();
+    GraphContext *gc = GraphContext_GetFromLTS();
+
+     /* Expanding the RETURN clause is a two phase operation:
+      * 1. Scan through every arithmetic expression within the original
+      * RETURN clause and add it to a temporary vector,
+      * if we bump into an asterisks marker indicating we should expand
+      * all nodes, relations and paths, add thoese to the temporary vector.
+      * 
+      * 2. Scanning though the temporary vector we expand each collapsed entity
+      * this will form the final RETURN clause. */
+
+    AST_ReturnNode *returnClause = ast->returnNode;
+    size_t returnElementCount = array_len(returnClause->returnElements);
+    AST_ReturnElementNode **elementsToExpand = array_new(AST_ReturnElementNode*, returnElementCount);
+
+    TrieMap *identifiers = NewTrieMap();
+    MatchClause_DefinedEntities(ast->matchNode, identifiers);
+    CreateClause_DefinedEntities(ast->createNode, identifiers);
+
+    // Scan through return elements, see if we can find '*' marker.
+    for(int i = 0; i < returnElementCount; i++) {
+        AST_ReturnElementNode *retElement = returnClause->returnElements[i];        
+        if(retElement->asterisks) {
+            // TODO: '*' can not be mixed with any other expression!
+            array_clear(elementsToExpand);
+            /* Expand with "RETURN *" queries.
+             * Extract all entities from MATCH clause and add them to RETURN clause.
+             * These will get expended later on. */
+
+            char *ptr;
+            tm_len_t len;
+            void *value;
+
+            TrieMapIterator *it = TrieMap_Iterate(identifiers, "", 0);
+            while(TrieMapIterator_Next(it, &ptr, &len, &value)) {
+                AST_GraphEntity *entity = (AST_GraphEntity*)value;
+                if(entity->anonymous) continue;
+
+                ptr[len] = '\0';
```

Comment:
I don't think it's safe to write into triemap pointers past `len-1`. Maybe memcpy onto a stack variable?